Makefile

```
@@ -55,7 +55,7 @@ ifeq ($(shell command -v ld.lld >/dev/null && uname -s),Linux)
 endif
 
 ifeq (${LLVM_CONFIG},)
-  $(error Could not locate llvm-config, make sure it is installed and in your PATH, or set LLVM_CONFIG)
+  $(error Could not locate llvm-config, or unrecognized version, make sure it is installed and in your PATH, or set LLVM_CONFIG)
```

Alternate suggestion:

```suggestion
  $(error Could not locate compatible llvm-config, make sure it is installed and in your PATH, or set LLVM_CONFIG)
```

---

Should we list the supported versions?

---

If there's an easy way to do that, it might be a good idea. But I suppose it would either require unnecessary complexity/dependencies or duplication. This is a rather low level developer tool, so I don't think it's very necessary
Maybe a reference to `src/llvm/ext/find-llvm-config` could be helpful. It's easy to look that up in the Makefile, though.

---

It's a static list, no? I don't mean the available supported versions, just all the supported versions. Could be a range like "Crystal supports LLVM 5-10". So it's just adding another place to keep updated.

---

> adding another place to keep updated. 

That's something I'd like to avoid because it's easy to miss out and such a list is completely useless when it's out of sync, so it's better to not have one.

---

Well "missing out" could be avoided if we start to have little runbook like things for tasks like this in our documentation. Another measure is to factor our the list of supported versions into a single file.